gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/DocumentReferenceTest.java

```diff
@@ -0,0 +1,173 @@
+        assertThatIdentifierIsValid(documentReference.getIdentifierFirstRep(), documentReference.getId());
+        assertThat(documentReference.getType().getText()).isEqualTo(NARRATIVE_STATEMENT_TYPE);
+        assertThat(documentReference.getAuthor().get(0).getReference()).isEqualTo("Practitioner/2D70F602-6BB1-47E0-B2EC-39912A59787D");
+        assertThat(documentReference.getDescription()).isEqualTo("Filename: 31B75ED0-6E88-11EA-9384-E83935108FD5_patient-attachment.txt");
+        assertThat(documentReference.getIndexedElement().getValue().toInstant().toString()).isEqualTo("2010-01-14T00:00:00Z");
+        assertThat(documentReference.getCreatedElement().getValue().toInstant().toString()).isEqualTo("2020-10-12T13:33:44Z");
+    }
+
+    private void assertOptionalValidData(DocumentReference documentReference) {
+        assertThat(documentReference.getId()).isEqualTo(NARRATIVE_STATEMENT_ROOT_ID);
+        assertThat(documentReference.getMeta().getProfile().get(0).getValue()).isEqualTo(META_PROFILE);
+        assertThat(documentReference.getStatus()).isEqualTo(Enumerations.DocumentReferenceStatus.CURRENT);
+        assertThatIdentifierIsValid(documentReference.getIdentifierFirstRep(), documentReference.getId());
+        assertThat(documentReference.getType().getText()).isEqualTo(NARRATIVE_STATEMENT_TYPE);
+        assertThat(documentReference.getAuthor().get(0).getReference()).isEqualTo("Practitioner/2D70F602-6BB1-47E0-B2EC-39912A59787D");
+        assertThat(documentReference.getDescription()).isNull();
+        assertThat(documentReference.getCreated()).isNull();
+        assertThat(documentReference.getContent()).isEmpty();
+    }
+
+    private void assertAttachmentData(DocumentReference documentReference) {
+        assertThat(documentReference.getContent().get(0).getAttachment().getTitle()).isEqualTo(FILENAME);
+        assertThat(documentReference.getContent().get(0).getAttachment().getUrl()).isEqualTo(URL);
+        assertThat(documentReference.getContent().get(0).getAttachment().getContentType()).isEqualTo(CONTENT_TYPE);
+    }
+
+    private void assertDocumentReferenceWithAbsentAttachment(DocumentReference documentReference) {
+        assertThat(documentReference.getId()).isEqualTo(NARRATIVE_STATEMENT_ROOT_ID);
+        assertThat(documentReference.getContent().get(0).getAttachment().getTitle()).isEqualTo(PLACEHOLDER);
+        assertThat(documentReference.getContent().get(0).getAttachment().getUrl()).isNull();
+        assertThat(documentReference.getContent().get(0).getAttachment().getContentType()).isEqualTo(CONTENT_TYPE);
+    }
+
+    private void assertThatIdentifierIsValid(Identifier identifier, String id) {
+        assertThat(identifier.getSystem()).isEqualTo(IDENTIFIER_SYSTEM);
+        assertThat(identifier.getValue()).isEqualTo(id);
+    }
+
+    private static Patient createPatient() {
+        Patient patient = new Patient();
+        patient.setId(randomUUID().toString());
```

Comment:
Dont set this as a random UUID, if you have a hard coded one you should be able to assert the subject is also valid